src/fmu/dataio/_filedata_provider.py

```diff
@@ -63,10 +64,11 @@ def __post_init__(self):
 
         self.fmu_context = self.dataio._usecontext  # may be None!
 
-        logger.info("Initialize %s", __class__)
+        logger.info("Initialize %s", self.__class__)
 
-    def derive_filedata(self):
+    def derive_filedata(self) -> None:
         relpath, symrelpath = self._get_path()
+        assert relpath is not None
```

Comment:
I think this can assert could be removed -- not that it changes functionality, but if `_get_path` is annotated to return `tuple[Path, Path] | tuple[Path, None]`